server/core/src/objDesc.cpp

```diff
@@ -28,10 +29,47 @@
 
 int
 initL1desc() {
-    memset( L1desc, 0, sizeof( L1desc ) );
+    for (auto&& e : L1desc) {
+        init_l1desc(e);
+    }
+
     return 0;
 }
 
+auto init_l1desc(l1desc& _l1d) -> void
+{
+    _l1d.l3descInx = 0;
+    _l1d.inuseFlag = 0;
+    _l1d.oprType = 0;
+    _l1d.openType = 0;
+    _l1d.oprStatus = 0;
+    _l1d.dataObjInpReplFlag = 0;
+    _l1d.copiesNeeded = 0;
+    _l1d.bytesWritten = 0;
+    _l1d.dataSize = 0;
+    _l1d.replStatus = 0;
+    _l1d.chksumFlag = 0;
+    _l1d.srcL1descInx = 0;
+    _l1d.remoteL1descInx = 0;
+    _l1d.stageFlag = 0;
+    _l1d.purgeCacheFlag = 0;
+    _l1d.lockFd = 0;
+
+    _l1d.dataObjInp = nullptr;
+    _l1d.dataObjInfo = nullptr;
+    _l1d.otherDataObjInfo = nullptr;
+    _l1d.replDataObjInfo = nullptr;
+    _l1d.remoteZoneHost = nullptr;
+
+    _l1d.pluginData.clear();
+    _l1d.replica_token.clear();
+
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    std::memset(_l1d.chksum, 0, sizeof(l1desc::chksum));
```

Comment:
Might be worth considering `std::fill` vs `std::memset`.

Reply:
I could use `std::fill`, but `std::memset` seems perfectly fine for this case given `l1desc::chksum` is a plain ole C array.

Is there an argument against using `std::memset`?

Reply:
After some discussion, I'm going to move forward with `std::memset`.

There are online articles that hint at the possibility that `std::fill` can produce the same assembly as `std::memset`, but not always.

We should investigate in which cases it is better to use one function over the other.